src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -3336,11 +3336,9 @@ namespace N
 {
     public record C(int P, bool B)
     {
-        public {|CS0111:{|CS8862:C|}|}(int p, bool b)
+        public {|CS0111:C|}(int p, bool b) : {|CS0121:this|}(p, b)
```

Comment:
I changed the way constructors with the same parameters as the primary constructor, but are not simple constructors, are handled. Previously the only one we considered was the first constructor with matching parameters, and all other constructors (even if they also had matching parameters) were considered non-primary. Now we look for the first constructor whose parameters match and is also a "simple" primary constructor, meaning we can delete it. This does mean that in this case (where the parameters match but it isn't "simple"), we treat it as a non-primary, which causes a signature error with the primary constructor.

The error changed from "Record constructors need to have a `this` initializer" to "I'm not sure which constructor you mean when you call `this` because there are two constructors with the same signature." I can add filtering so that if the constructor has the exact same signature as the primary, we just ignore it entirely.

Reply:
@Soreloser2 Ah, I don't think I was necessarily saying the old behavior was bad; I was just trying to understand it. Do you think the newer behavior here is better?

Reply:
I added the filtering so that the original behavior (2 errors on `C`, don't create `this` initializer) happens whenever we see a constructor whose parameters conflict with our primary, but isn't a simple primary constructor that we can delete.

I think this way with the original behavior was slightly better, only because it didn't make any changes to the constructor. I think it's easier for the user to understand that creating a positional record introduced the constructor signature conflict, rather than something that we broke by changing their constructor.